### PR TITLE
Dockerfile.ui: improve cacheability

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

Since Docker doesn't know which programs use which environment variables, any `RUN` steps that follow an `ARG` step will not be loaded from the cache if the value of the `ARG` changes. In this case, the `yarn` command will not be loaded from the cache if any of the `ARG` values change, even though (to my knowledge) it does not use any of those values.

Fix this by moving the `ARG` steps as far down as possible.

### How has this been tested?

I ran `docker build` with various values of `SOURCE_MAPS_TOKEN` and observed which steps were loaded from cache.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **Chores**
	- Updated the Dockerfile for improved handling of build arguments, enhancing the build process for the UI.
	- Streamlined the declaration and usage of build arguments to ensure better control during the UI build execution.

### CODE REVIEW

File: Dockerfile.ui [hunk 1]
```diff
@@ -29,6 +23,13 @@ COPY cvat-core/ /tmp/cvat-core/
 COPY cvat-canvas3d/ /tmp/cvat-canvas3d/
 COPY cvat-canvas/ /tmp/cvat-canvas/
 COPY cvat-ui/ /tmp/cvat-ui/
+
+ARG WA_PAGE_VIEW_HIT
```
Comment (bsekachev) [inline Dockerfile.ui:27]: I think any mentioning of WA_PAGE_VIEW_HIT may be removed as we dropped google web analytics long time ago 

@azhavoro I am correct?

Comment (SpecLad) [reply Dockerfile.ui:27]: FWIW, this variable is still used in the code.

Comment (bsekachev) [reply Dockerfile.ui:27]: It is used in .ts code that also needs to be removed

Comment (bsekachev) [reply Dockerfile.ui:27]: This code will work if `WA_PAGE_VIEW_HIT`, but it is not defined anywhere in saas repo as I can see

Comment (bsekachev) [reply Dockerfile.ui:27]: Okay, I will send one more PR to solve it

**Review by bsekachev — APPROVED** [review]

**Pull request merged**